WordPress/gutenberg
# Navigation block: Create classic menus empty

## What?
This is an attempt to fix #42850

## Why?
When you create a new navigation menu from a classic menu then a new wp_navigation CPT is created, which is already populated with the content of the classic navigation. The upshot of this is that as soon as the classic navigation is created the fallback behaviour of the block is changed and the classic menu is used instead of page list (as described in https://github.com/WordPress/gutenberg/issues/42799)

## How?
When we create a new "classic" wp_navigation CPT we do it without any innerblocks, so its empty. This means that the fallback behaviour is unchanged. We then populate the inner blocks with the navigation items that were returned from the API. This should mark the block as "dirty" for the entity saving flow, so it feels like it should be a simpler approach. 

## Testing Instructions

To replicate the issue

1. Remove all navigation menus from wp-admin/edit.php?post_type=wp_navigation
2. Ensure you have a classic menu (you will need to switch to a classic theme to create one if you don't have one)
3. Create a navigation block without any inner blocks (if you're using a navigation block in a theme you'll probably need to remove the blocks from the template).
4. Check that you see a page list fallback on your site
5. Select the block and click on "Classic menus > primary`, or whatever it is called.
6. Now reload the frontend of your site and check that the fallback has changed to the classic menu.

With this PR enabled
6. The fallback is unchanged
7.  (not working from this point) You should see the classic menu inner blocks in the editor
8. When you go to save you have the option to save the new CPT
9. When you have saved the frontend now displays the classic navigation menu items.
